tests/components/mqtt/test_init.py

```diff
@@ -2097,3 +2283,32 @@ async def test_service_info_compatibility(hass, caplog):
     with patch("homeassistant.helpers.frame._REPORTED_INTEGRATIONS", set()):
         assert discovery_info["topic"] == "tasmota/discovery/DC4F220848A2/config"
     assert "Detected integration that accessed discovery_info['topic']" in caplog.text
+
+
+async def test_subscribe_connection_status(hass, caplog, mqtt_mock, mqtt_client_mock):
+    """Test connextion status subscription."""
+
+    @callback
+    async def async_mqtt_connected(status: bool) -> bool:
+        """Update state on connection/disconnection to MQTT broker."""
+        _LOGGER.info("connection status %s", status)
```

Comment:
Add the calls to a list or something, instead of verifying the calls by checking for prints.

```py
nonlocal mqtt_connected_calls
mqtt_connected_calls.append(status)
```

Reply:
done